src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -305,10 +317,25 @@ public function setup(): void
 
         $this->exchange()->declareExchange();
 
-        $this->queue()->declareQueue();
-        $this->queue()->bind($this->exchange()->getName(), $this->queueConfiguration['routing_key'] ?? null);
+        foreach ($this->queuesConfiguration as $queueName => $queueConfig) {
+            $this->queue($queueName)->declareQueue();
+            foreach ($queueConfig['routing_keys'] ?? [] as $routingKey) {
```

Comment:
@weaverryan , don't forget to rename `routing_keys` to `binding_keys` for this binding part (also in constructor doc)